src/lib/background/domains/status/preset.js

```diff
@@ -9,14 +9,39 @@ const {
 const presets = require("../../../../data/domains");
 
 /**
- * Resolve flattr status for given URL or domain
+ * Retrieve only tree nodes which are relevant to the given host
+ * @param {string[]} hostParts - host parts (e.g. ["example", "com"])
+ * @return {any[]} tree nodes corresponding to given host parts
+ */
+function filterTree(hostParts)
+{
+  let treeNodes = [];
+  let tree = presets.status;
+  hostParts = hostParts.slice().reverse();
+
+  for (let hostPart of hostParts)
+  {
+    tree = tree[hostPart];
+    if (typeof tree == "undefined")
+      break;
+
+    treeNodes.push(tree);
```

Comment:
nit using `unshift` here instead of `push` would allow us to avoid having to reverse the return value.

Reply:
Good point. Done.